state/store.go

```diff
@@ -36,10 +36,47 @@ func calcABCIResponsesKey(height int64) []byte {
 	return []byte(fmt.Sprintf("abciResponsesKey:%v", height))
 }
 
+//----------------------
+
+type StoreI interface {
```

Comment:
```suggestion
type Store interface {
```

Reply:
One thing to keep in mind here is that any change to an interface (including adding a method) is a breaking change in Go. If we're planning to expose the state store in any public APIs (which I'm guessing we will at some point), then we should strongly consider using a struct instead, which gives us more flexibility.